Bottom: 4fcd5b1b6e9f876c739e9aba4cb2b2ad9c38e1e6
Top:    a3988a5bc55f6781f9f3e69c59b0896af194a3aa
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 00:19:09 +0100

draw.c: Split "Gathering" from real drawing routines (Element)


---

diff --git a/src/draw.c b/src/draw.c
index 7e8690f..2aeefc1 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -54,6 +54,9 @@
 #include <dmalloc.h>
 #endif
 
+#undef NDEBUG
+#include <assert.h>
+
 #ifndef MAXINT
 #define MAXINT (((unsigned int)(~0))>>1)
 #endif
@@ -1491,6 +1494,8 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
 void
 DrawElement (ElementTypePtr Element)
 {
+  assert (Gathering);
+
   DrawElementPackage (Element);
   DrawElementName (Element);
   DrawElementPinsAndPads (Element);
@@ -1823,6 +1828,14 @@ DrawObject (int type, void *ptr1, void *ptr2)
     }
 }
 
+static void
+draw_element (ElementTypePtr element)
+{
+  DrawElementPackage (element);
+  DrawElementName (element);
+  DrawElementPinsAndPads (element);
+}
+
 /* ---------------------------------------------------------------------------
  * HID drawing callback.
  */
@@ -1850,7 +1863,7 @@ hid_expose_callback (HID * hid, BoxType * region, void *item)
   if (item)
     {
       doing_pinout = true;
-      DrawElement ((ElementTypePtr)item);
+      draw_element ((ElementType *)item);
       doing_pinout = false;
     }
   else
